Testcases + Docker + CircleCI integration

purukaushik opened the pull request:
BGZL - meta-task -1286367
  CircleCI integration- 1286936; 
 dockerize- 1286938; 
 test deploy on ec2 tasks - 1286940
and - test cases - 1286619

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/purukaushik/telemetry-schema-services/3)

<!-- Reviewable:end -->

purukaushik committed:
adding CircleCI yml

purukaushik committed:
Adding tests for commons services

purukaushik committed:
renaming packages to pip requirements.txt file

purukaushik committed:
adding nose2 to test in circle.yml

purukaushik committed:
Adding little button for CircleCI

purukaushik committed:
Test file for testing the flask service app

purukaushik committed:
adding docker in circle.yml : "may not actually work" commit

purukaushik committed:
Adding tests for different /schema/ endpoints

purukaushik committed:
commenting docker stuff for now

purukaushik committed:
Dockerize...strategory

purukaushik committed:
adding dockerization to circle.yml

purukaushik committed:
Pointing docker to testcases github branch instead of master for now

purukaushik committed:
Adding file upload test cases

purukaushik committed:
Fixing test case on invalid file upload scenario

purukaushik committed:
docker pushing logic: try

purukaushik committed:
testing: remove file log

purukaushik committed:
testing: remove emacs, nano from Dockerfile apt install

purukaushik committed:
token docker deploy script for ec2

purukaushik committed:
syslog related. not needed. will use mozlog

purukaushik committed:
Stdout logging with mozlogy logging

Dockerfile test changes

Changes for mozlog cloud services logger pip install

refactoring imports; removing upload folder unwanted stuff

removing extra references

mreid-moz commented:
r? @maurodoglio 

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-:-KNDIYXOiSHDGhATWRgr:-1004198818)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
Merge mozlog; remove mozlog branch git checkout

purukaushik committed:
Dockerfile refactor

purukaushik commented:
@mreid-moz , @maurodoglio  I am making changes to this branch to integrate the mozlogging package

purukaushik committed:
remove pwd in dockerfile

purukaushik committed:
run nose2 from inside the docker container

purukaushik committed:
updating status badge for testcases branch

purukaushik committed:
remove docker info.

maurodoglio commented:
A few general comments:
- The project is missing a top level python module. See [the Hitchhiker’s Guide to Python](http://docs.python-guide.org/en/latest/writing/structure/#structuring-your-project) for more info
- One of the consequences of the previous point is that you are shuffling around a logger object to keep all the app logging under a unique namespace. This is not needed. A good practice is to have one logger per module (identified by  __name__ ) so that all the loggers will be automatically namespaced under the top level module.
- Given that the project is at a very early stage, it would be super useful to run a linter (flake8 is a good one) on CI, so that style errors get caught as soon as possible (ideally before the code review process)

---

Reviewed 6 of 10 files at r1, 1 of 2 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[circle.yml, line 16 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNH54We52yDavYvV0o4:-KNH54WfWggQnjzguurK:1291044368) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/circle.yml#L16)):*
> ```YAML
>     commands:
>       - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
>       - docker push purush/telemetry-schema-services
> ```

I don't have any experience with circleci, so this is gonna be a rubber stamp. 

---

*[Dockerfile, line 9 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNGp0W-paQmdWXyntzb:-KNGp0W0Pmrc4y2YGFMR:1490167258) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/Dockerfile#L9)):*
>  apt update it
> RUN apt-get update
> # get essentials
> RUN apt-get install -y tar git curl wget dialog net-tools build-essential
> # get python

Please put these commands in the same RUN invocation (see https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/minimize-the-number-of-layers)

---

*[Dockerfile, line 11 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNH5jeVKAcpMbAHhijR:-KNH5jeWJbGf4TZU-pyR:1639880463) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/Dockerfile#L11)):*
> ```Dockerfile
> RUN apt-get install -y python python-dev python-distribute python-pip
> # clone our repo
> #RUN git clone https://github.com/purukaushik/telemetry-schema-services.git
> ```

Why do you need to clone the repo instead of adding the local folder? 

---

*[Dockerfile, line 19 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNGpfCQmnt4PCDyuL40:-KNGpfCR2OdM5xzYTUbz:1137704043) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/Dockerfile#L19)):*
> mozilla-cloud-services-logger
Is there a reason why this is not fetched from pypi? In any case, this should be listed in the requirements file even if the source is git. See https://pip.pypa.io/en/stable/reference/pip_install/#git

---

*[Dockerfile, line 26 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNGxukhjqz4-jvi1RL4:-KNGxukhjqz4-jvi1RL5:-87643976) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/Dockerfile#L26)):*
> ```Dockerfile
> WORKDIR /telemetry-schema-services
> 
> CMD /telemetry-schema-services/mozschemas_service.py -p 8080 --host='0.0.0.0'
> ```

If I understand correctly this will execute app.run even in a production environment, which [is not ideal](http://flask.pocoo.org/docs/0.11/api/#flask.Flask.run)
I would suggest to use a proper production-ready webserver like gunicorn. 

---

*[mozschemas_service.py, line 27 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNHDui8WvWSwCLBy8Qs:-KNHDui9HhX_sSn_zvMb:-1817768572) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/mozschemas_service.py#L27)):*
> ```Python
> from docopt import docopt
> from mozilla_cloud_services_logger.formatters import JsonLogFormatter
> import sys
> ```

Please follow the [pep8 recommendations](https://www.python.org/dev/peps/pep-0008/#imports) for imports order

---

*[requirements.txt, line 5 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNGznjKrLsBnFTOKD3R:-KNGznjL5rtgleDrMwYP:-1200831955) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/requirements.txt#L5)):*
> ```
> gitpython
> docopt
> nose2
> ```

This should be in a separate TEST_ or DEV_ requirements file as it's not required to run the app.

---

*[test_config.json, line 1 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNH2CPt-HRdLhjOtg-q:-KNH2CPt-HRdLhjOtg-r:153665087) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/test_config.json#L1)):*
> ```JSON
> {
> ```

Shouldn't this file be moved to tests/config.json ?

---

*[tests/test_commons.py, line 23 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNHF0kuWoO93nh-vu8Y:-KNHF0kvduBD9uGSjPEe:1931246442) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/tests/test_commons.py#L23)):*
> ```Python
>         namespace = self.config["namespace"]
>         docTypes = self.config["docTypes"]
>         result = self.helper.get_doctypes_versions(namespace=namespace,docType=None,logger=self.logger)
> ```

whitespace between parameters please

---

*[tests/test_commons.py, line 63 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNH2hX8XitQYzgoMOUU:-KNH2hX8XitQYzgoMOUV:1705301420) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/tests/test_commons.py#L63)):*
> if __name__ == '__main__':

I don't think this is required in order to run the tests. The test runner (nose?) should take care of discovery/execution of TestCases

---

*[tests/test_service.py, line 91 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNHFcgnh7hfHfBKtfLk:-KNHFcgoKjYcy_l9Pvxs:1727578530) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/tests/test_service.py#L91)):*
> ```Python
>         self.assertEquals(resp.status_code, status_code)
>         # check mimetype=application/json
>         self.assertEquals(resp.mimetype, mimetype)
> ```

files should finish with a blank line when possible

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-:-KNHElCj-AnkShIWzdBx:1558362985)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
Merge branch 'testcases' of github.com:purukaushik/telemetry-schema-services into testcases

purukaushik commented:
Review status: 8 of 9 files reviewed at latest revision, 11 unresolved discussions.

---

*[tests/test_commons.py, line 23 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNHF0kuWoO93nh-vu8Y:-KNZ8lThhtXRiCKbIR_7:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/tests/test_commons.py#L23)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> whitespace between parameters please
</details>
Done.

---

*[tests/test_commons.py, line 63 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNH2hX8XitQYzgoMOUU:-KNZ8aTZaN97Mtx2J8eD:-460928462) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/tests/test_commons.py#L63)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> > if __name__ == '__main__':
> 
> I don't think this is required in order to run the tests. The test runner (nose?) should take care of discovery/execution of TestCases
</details>
Done. Removed main

---

*[tests/test_service.py, line 91 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNHFcgnh7hfHfBKtfLk:-KNZ8Z2Zm1OIjqsE_VJD:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/tests/test_service.py#L91)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> files should finish with a blank line when possible
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
Little refactoring changes from mdoglio comments.

purukaushik commented:
Review status: 8 of 9 files reviewed at latest revision, 11 unresolved discussions.

---

*[circle.yml, line 16 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNH54We52yDavYvV0o4:-KNZAouWzUVe1UqFiyxx:1923409614) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/circle.yml#L16)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> I don't have any experience with circleci, so this is gonna be a rubber stamp. 
</details>
Oh. Ok. I am following recommendations from the clourservices mana page for docker based deployments.

---

*[mozschemas_service.py, line 27 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNHDui8WvWSwCLBy8Qs:-KNZA_l802OGG8YoHd9w:-1077034101) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/mozschemas_service.py#L27)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Please follow the [pep8 recommendations](https://www.python.org/dev/peps/pep-0008/#imports) for imports order
</details>
Going to refactor whole project with pep8 formatting after fixing the major comments. :thumbsup: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
test_config.json -> config.json changes

test_config.json -> tests/config.json

test_config.json -> tests/config.json

purukaushik commented:
Review status: 6 of 9 files reviewed at latest revision, 11 unresolved discussions.

---

*[test_config.json, line 1 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNH2CPt-HRdLhjOtg-q:-KNZSdLdQ2tAqhngjmnJ:-660737743) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/test_config.json#L1)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Shouldn't this file be moved to tests/config.json ?
</details>
Yeah. Sure. :thumbsup:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
Squashed review fixes + refactoring

adding check-out logic to CLI. renaming __init__.py, trailing slash rule, extract helper object, git_checkout on startup

spaces refactoring

Checkpoint: made commons access git_config.json&quot;s os_dir property instead of CWD based one

fixed path problems with git_checkouts

Fixing Dockerfile review comments except Gunicornize

Gunicorize flask

purukaushik committed:
remove pwd from Dockerfile

purukaushik commented:
Review status: 2 of 15 files reviewed at latest revision, 9 unresolved discussions.

---

*[Dockerfile, line 9 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNGp0W-paQmdWXyntzb:-KNiOVuFjV0d71eETzuX:158594303) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/Dockerfile#L9)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> >  apt update it
> > RUN apt-get update
> > # get essentials
> > RUN apt-get install -y tar git curl wget dialog net-tools build-essential
> > # get python
> 
> Please put these commands in the same RUN invocation (see https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/minimize-the-number-of-layers)
</details>
done :thumbsup: 

---

*[Dockerfile, line 11 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNH5jeVKAcpMbAHhijR:-KNiOZNgmI_XQ0iGWfWh:1128598956) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/Dockerfile#L11)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Why do you need to clone the repo instead of adding the local folder? 
</details>
removed cloning. made it just add current folder stuff into docker container

---

*[Dockerfile, line 19 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNGpfCQmnt4PCDyuL40:-KNj95hIdSOsckbokoOq:-1983683590) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/Dockerfile#L19)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> > mozilla-cloud-services-logger
> Is there a reason why this is not fetched from pypi? In any case, this should be listed in the requirements file even if the source is git. See https://pip.pypa.io/en/stable/reference/pip_install/#git
</details>
added this to requirements.txt

---

*[Dockerfile, line 26 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3#-KNGxukhjqz4-jvi1RL4:-KNj98_vG1m0RUacEz6M:-1531951923) ([raw file](https://github.com/purukaushik/telemetry-schema-services/blob/86712955456db55bfea934116a5fe0453daafff0/Dockerfile#L26)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> If I understand correctly this will execute app.run even in a production environment, which [is not ideal](http://flask.pocoo.org/docs/0.11/api/#flask.Flask.run)
> I would suggest to use a proper production-ready webserver like gunicorn. 
</details>
gunicornized! :thumbsup:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-services/3)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
Switching git schema repo to mreid-moz:commons

purukaushik committed:
Renaming repository - side-effect changes

purukaushik committed:
Removing cloning for services logger

purukaushik committed:
app changes for cloud ops dockerflow requirement

purukaushik committed:
Adding top level module

purukaushik committed:
Big Change - adding module level loggers, tests

purukaushik commented:
fixed - top level module+ module level logging change

---

Review status: 2 of 20 files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-:-KNrnzjb0QJAE4T_IvfA:1162149013)*
<!-- Sent from Reviewable.io -->

maurodoglio commented:
Reviewed 1 of 12 files at r1, 2 of 5 files at r4, 3 of 15 files at r5, 2 of 5 files at r6, 19 of 19 files at r7.
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[__init__.py, line 0 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO4ydTnY87Kq0ylY0dR:-KO4ydTnY87Kq0ylY0dS:1793231439) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/__init__.py#L0)):*
Please delete this file from the root folder. As per the previous comments, the root folder shouldn't be a python module.

---

*[circle.yml, line 15 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO4yy8w1cFSQxUfmiHz:-KO4yy8x38o8FIsRk8Xf:-500593164) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/circle.yml#L15)):*
> ```YAML
>     commands:
>       - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
>       - docker push purush/telemetry-schema-service
> ```

Please add a new line at the end of each file where possible

---

*[Dockerfile, line 19 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO4zX24bKREk3uIqHcN:-KO4zX25mBVz9ZwvmWJ-:1164401347) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/Dockerfile#L19)):*
> ```Dockerfile
> 
> # make mozschemas_service.py runnable
> RUN chmod +x /telemetry-schema-service/app/mozschemas_service.py
> ```

Now that you are using gunicorn there's no need to make this file runnable

---

*[requirements.txt, line 6 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5-_PJhJvV6vfC3DWC:-KO5-_PKZuXhFzSrxFyF:807657125) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/requirements.txt#L6)):*
> ```
> docopt
> nose2
> git+https://github.com/mozilla/mozilla-cloud-services-logger.git#egg=mozilla-cloud-services-logger
> ```

This package is available on pypi, please install it from there.

---

*[requirements.txt, line 7 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5009m0J40-7qvCj3B:-KO5009m0J40-7qvCj3C:-500593164) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/requirements.txt#L7)):*
> ```
> nose2
> git+https://github.com/mozilla/mozilla-cloud-services-logger.git#egg=mozilla-cloud-services-logger
> gunicorn
> ```

Please add a new line at the end of each file where possible

---

*[version.json, line 1 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO50NcZ2BPJrNUrGyQn:-KO50NcZ2BPJrNUrGyQo:-1808971771) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/version.json#L1)):*
> ```JSON
> {
> ```

What is this file for? Are you sure this needs to be under version control (instead of being automatically generated somehow)?

---

*[version.json, line 6 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO50BBi20pkWj6lIQ7n:-KO50BBi20pkWj6lIQ7o:-500593164) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/version.json#L6)):*
> ```JSON
>   "commit" : "2bb97482445819ef3750d631201abc9b3b7936ce",
>   "build"  : "unknown"
> }
> ```

Please add a new line at the end of each file where possible

---

*[app/git_checkout.py, line 33 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO51QcrPZjzzt_hmlWk:-KO51QcsnAS1ub2dBh65:-1069219724) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/git_checkout.py#L33)):*
> ```Python
>             logger.error(err)
>             logger.warn('Git error while cloning. Using local')
>             pass
> ```

Please get rid of these 2 logger calls and use  logger.exception instead: 
```python
logger.exception('Git error while cloning. Using local')
```
Also, you can remove the pass statement is redundant in this context

---

*[app/git_checkout.py, line 53 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO546P-6vCWalvVSe9D:-KO546P0JRBz9e0H0TjC:1550815997) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/git_checkout.py#L53)):*
> ```Python
>         pass
>     except git.exc.GitCommandError:
>         logger.warn("Git error while pulling latest. using local")
> ```

Same as above, please use logger.exception and remove the redundant pass statement

---

*[app/git_checkout.py, line 65 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO55icn0ZWAGBwFw__U:-KO55icn0ZWAGBwFw__V:-230782279) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/git_checkout.py#L65)):*
> ```Python
> 
> if __name__ == '__main__':
>     gitcheckout()
> ```

I thought the entrypoint to gitcheckout is the mozschemas_cli module, are you sure you need this?

---

*[app/mozschemas_cli.py, line 63 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO56WBBTq8HtjgyPnn_:-KO56WBC8jL5OwRPbTJR:-500593164) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/mozschemas_cli.py#L63)):*
> ```Python
>     else:
>         print "ERROR! Specify Namespace."
>         print __doc__
> ```

Please add a new line at the end of each file where possible

---

*[app/mozschemas_service.py, line 15 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5Fn_9Z0g9yRX-Z8op:-KO5Fn_9Z0g9yRX-Z8oq:-1516646015) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/mozschemas_service.py#L15)):*
> """Schema Service Core.
> Starts the core python schema hosting/validation web service API.
> Usage:
> mozschemas_service.py [-p <port>] [--host=<host>]
> mozschemas_service.py (-h | --help)
> 
> Options:
> -h --help       Show this screen.
> --host=<host>   Hostname [default: 0.0.0.0]
> -p <port>       Port number to run flask on [default: 8080]

If you use gunicorn for development you can get rid of these docstrings as well

---

*[app/mozschemas_service.py, line 196 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5B-2M3DOrK2kt8Aau:-KO5B-2M3DOrK2kt8Aav:-1737860894) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/mozschemas_service.py#L196)):*
> ```Python
>     port = arguments.get('-p', 8080)
> 
>     app.run(host=host, port=port, threaded=True)
> ```

Now that you have gunicorn in the requirements, why not use it for development as well? At the end of the day it's less code to maintain.

---

*[tests/test_commons.py, line 18 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5A00KMeluS3pB3Fur:-KO5A00KMeluS3pB3Fus:1307908748) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/tests/test_commons.py#L18)):*
> ```Python
> 
>         self.helper = SchemasLocalFilesHelper()
>         self.logger = getLogger(__name__)
> ```

why do you need a logger for this test case?

---

*[tests/test_commons.py, line 19 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO58_seuAQ5c-Fh0_7q:-KO58_seuAQ5c-Fh0_7r:335307216) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/tests/test_commons.py#L19)):*
> ```Python
>         self.helper = SchemasLocalFilesHelper()
>         self.logger = getLogger(__name__)
>         __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
> ```

why did you surround location with '__' ? In python that usually indicates either a magic variable or method (like __name__ for example in the line above).

---

*[tests/test_service.py, line 72 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5BiHPS6mmjpEbltXp:-KO5BiHPS6mmjpEbltXq:1264299207) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/tests/test_service.py#L72)):*
> ```Python
> 
>                 if docType not in self.config['skip_invalid']:
>                     __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
> ```

Again, please don't define variables surrounded by double underscores.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3)*
<!-- Sent from Reviewable.io -->

purukaushik commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[__init__.py, line  \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO4ydTnY87Kq0ylY0dR:-KO5fWRhW_NyYL2NKBM4:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/__init__.py#L)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Please delete this file from the root folder. As per the previous comments, the root folder shouldn't be a python module.
</details>
Done.

---

*[circle.yml, line 15 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO4yy8w1cFSQxUfmiHz:-KO5fXP043xCdkIllZjn:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/circle.yml#L15)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Please add a new line at the end of each file where possible
</details>
Done.

---

*[Dockerfile, line 19 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO4zX24bKREk3uIqHcN:-KO5g72HogGKcN2d2ONu:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/Dockerfile#L19)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Now that you are using gunicorn there's no need to make this file runnable
</details>
Done.

---

*[requirements.txt, line 6 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5-_PJhJvV6vfC3DWC:-KO5g3R8ykE5afK8ZOEg:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/requirements.txt#L6)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> This package is available on pypi, please install it from there.
</details>
Done.

---

*[requirements.txt, line 7 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5009m0J40-7qvCj3B:-KO5g3uUX_SSxrzcKpoS:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/requirements.txt#L7)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Please add a new line at the end of each file where possible
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
First round of review changes

purukaushik committed:
end of line in requirements file

purukaushik committed:
Review fixes 8/1.

purukaushik commented:
Review status: 11 of 20 files reviewed at latest revision, 17 unresolved discussions.

---

*[requirements.txt, line 5 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KNGznjKrLsBnFTOKD3R:-KO5l1yhkMoP48ZPpgIW:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/86712955456db55bfea934116a5fe0453daafff0/requirements.txt#L5)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> This should be in a separate TEST_ or DEV_ requirements file as it's not required to run the app.
</details>
Done.

---

*[version.json, line 1 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO50NcZ2BPJrNUrGyQn:-KO5irmZxhyHc6ZEF2cI:1221540473) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/version.json#L1)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> What is this file for? Are you sure this needs to be under version control (instead of being automatically generated somehow)?
</details>
robotblake is gonna throw in a PR to generate this automatically. Will remove once that script is in place.

---

*[version.json, line 6 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO50BBi20pkWj6lIQ7n:-KO5j4YoIITcMbkwhVjJ:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/version.json#L6)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Please add a new line at the end of each file where possible
</details>
Done.

---

*[app/git_checkout.py, line 33 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO51QcrPZjzzt_hmlWk:-KO5ksnuwhbLIG7Iu-Nk:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/git_checkout.py#L33)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Please get rid of these 2 logger calls and use  logger.exception instead: 
> ```python
> logger.exception('Git error while cloning. Using local')
> ```
> Also, you can remove the pass statement is redundant in this context
</details>
Done.

---

*[app/git_checkout.py, line 53 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO546P-6vCWalvVSe9D:-KO5kuEuTUM2_D1b5wYT:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/git_checkout.py#L53)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Same as above, please use logger.exception and remove the redundant pass statement
</details>
Done.

---

*[app/git_checkout.py, line 65 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO55icn0ZWAGBwFw__U:-KO5k_jPfw0QNWLxLIuQ:845602236) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/git_checkout.py#L65)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> I thought the entrypoint to gitcheckout is the mozschemas_cli module, are you sure you need this?
</details>
I am having it in case we need a cron job in the future to just pull the mozilla-pipeline-schemas repository without calling the CLI.

---

*[app/mozschemas_cli.py, line 63 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO56WBBTq8HtjgyPnn_:-KO5kCbQbwfMjDOWTvG5:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/mozschemas_cli.py#L63)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Please add a new line at the end of each file where possible
</details>
Done.

---

*[app/mozschemas_service.py, line 15 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5Fn_9Z0g9yRX-Z8op:-KO5je7A4XLV_NdPI6xy:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/mozschemas_service.py#L15)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> > """Schema Service Core.
> > Starts the core python schema hosting/validation web service API.
> > Usage:
> > mozschemas_service.py [-p <port>] [--host=<host>]
> > mozschemas_service.py (-h | --help)
> > 
> > Options:
> > -h --help       Show this screen.
> > --host=<host>   Hostname [default: 0.0.0.0]
> > -p <port>       Port number to run flask on [default: 8080]
> 
> If you use gunicorn for development you can get rid of these docstrings as well
</details>
Done.

---

*[app/mozschemas_service.py, line 196 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5B-2M3DOrK2kt8Aau:-KO5jcSsO9mINALTFHKr:1660675680) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/mozschemas_service.py#L196)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Now that you have gunicorn in the requirements, why not use it for development as well? At the end of the day it's less code to maintain.
</details>
Sure. Done.

---

*[tests/test_commons.py, line 18 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5A00KMeluS3pB3Fur:-KO5k44amoF9aPyW9zXi:-1106195150) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/tests/test_commons.py#L18)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> why do you need a logger for this test case?
</details>
Done. Not needed. Spillover from earlier passing around of logger instances.

---

*[tests/test_commons.py, line 19 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO58_seuAQ5c-Fh0_7q:-KO5k8jEHI2OKdUtQsvE:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/tests/test_commons.py#L19)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> why did you surround location with '__' ? In python that usually indicates either a magic variable or method (like __name__ for example in the line above).
</details>
Done.

---

*[tests/test_service.py, line 72 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO5BiHPS6mmjpEbltXp:-KO5k3JwwWCrQC_35dvD:-499146009) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/tests/test_service.py#L72)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> Again, please don't define variables surrounded by double underscores.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
Reverting back to one requirements.txt. Broke CircleCI

maurodoglio commented:
Fix the last 2 nits and this wil be a <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 1 of 12 files at r1, 3 of 19 files at r7, 9 of 9 files at r8.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[requirements.txt, line 5 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KNGznjKrLsBnFTOKD3R:-KO9iyS_Nv07BdIuKAdA:1181175239) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/86712955456db55bfea934116a5fe0453daafff0/requirements.txt#L5)):*
<details><summary><i>Previously, purukaushik (Purush Swaminathan) wrote…</i></summary>
> Done.
</details>
I still see nose2 here. If there is anything unclear please let me know.

---

*[version.json, line 6 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO50BBi20pkWj6lIQ7n:-KO9jW8JDfeuoQVLMI81:-871705666) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/version.json#L6)):*
<details><summary><i>Previously, purukaushik (Purush Swaminathan) wrote…</i></summary>
> Done.
</details>
I think the new line is still missing, unless reviewable is lying

---

*[app/git_checkout.py, line 65 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO55icn0ZWAGBwFw__U:-KO9huDrDwEtLYu2bt1X:802742353) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/git_checkout.py#L65)):*
<details><summary><i>Previously, purukaushik (Purush Swaminathan) wrote…</i></summary>
> I am having it in case we need a cron job in the future to just pull the mozilla-pipeline-schemas repository without calling the CLI.
</details>
I would still remove it, as per the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) principle.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-:-KO9jjdOTPxG8DvoCMmg:-918491894)*
<!-- Sent from Reviewable.io -->

purukaushik commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[requirements.txt, line 5 \[r3\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KNGznjKrLsBnFTOKD3R:-KOAkLPPpFpekScgT_y5:-1288249527) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/86712955456db55bfea934116a5fe0453daafff0/requirements.txt#L5)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> I still see nose2 here. If there is anything unclear please let me know.
</details>
I fixed this and then reverted the change as it broke circleCI as circle builds the project into a docker container and runs nose2 on the docker container. :confused:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3)*
<!-- Sent from Reviewable.io -->

purukaushik commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[version.json, line 6 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO50BBi20pkWj6lIQ7n:-KOAlAgjp7WSIEZQxObp:226907090) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/version.json#L6)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> I think the new line is still missing, unless reviewable is lying
</details>
Oh, snapple. Missed adding cause this file will be generated from Blake's script

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
YAGNIfy git_checkout, adding new line in version.json

purukaushik commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[app/git_checkout.py, line 65 \[r7\]](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-KO55icn0ZWAGBwFw__U:-KOAo1E4UquYx0UJA_ny:1665877846) ([raw file](https://github.com/purukaushik/telemetry-schema-service/blob/abb1eba7512acf94f5e44ab6ecd4fe193ac4fe0e/app/git_checkout.py#L65)):*
<details><summary><i>Previously, maurodoglio (Mauro Doglio) wrote…</i></summary>
> I would still remove it, as per the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) principle.
</details>
YAGNIfied! :smile:

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3)*
<!-- Sent from Reviewable.io -->

purukaushik committed:
YAGNIfy git_checkout, adding new line in version.json

purukaushik commented:
@maurodoglio Nits done, can we move the requirements.txt change to a patch  and move on with this for now ??

---

Review status: 17 of 19 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-:-KOB24NEILVYuy7Q6ge5:959276379)*
<!-- Sent from Reviewable.io -->

maurodoglio commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 19 files at r7, 2 of 2 files at r9.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/purukaushik/telemetry-schema-service/3#-:-KOBF3-c9jEJIv8hckvw:1417088721)*
<!-- Sent from Reviewable.io -->
